object_store/src/aws.rs

```diff
@@ -260,6 +270,56 @@ impl ObjectStoreApi for AmazonS3 {
         Ok(GetResult::Stream(s))
     }
 
+    async fn head(&self, location: &Path) -> Result<ObjectMeta> {
+        let key = location.to_raw().to_string();
+        let head_request = rusoto_s3::HeadObjectRequest {
+            bucket: self.bucket_name.clone(),
+            key: key.clone(),
+            ..Default::default()
+        };
+        let s = self
+            .client()
+            .await
+            .head_object(head_request)
+            .await
+            .map_err(|e| match e {
+                rusoto_core::RusotoError::Service(rusoto_s3::HeadObjectError::NoSuchKey(_)) => {
+                    Error::NotFound {
+                        path: key.clone(),
+                        source: e.into(),
+                    }
+                }
+                rusoto_core::RusotoError::Unknown(h) if h.status.as_u16() == 404 => {
+                    Error::NotFound {
+                        path: key.clone(),
+                        source: "resource not found".into(),
+                    }
+                }
+                _ => Error::UnableToHeadData {
+                    bucket: self.bucket_name.to_owned(),
+                    path: key.clone(),
+                    source: e,
+                },
+            })?;
+
+        // Note: GetObject and HeadObject return a different date format from ListObjects
```

Comment:
Maybe worth a ticket / url reference and examples

This is 🤯  lol

```suggestion
        // Note: GetObject and HeadObject return a different date format from ListObjects
        //
        // S3 List returns timestamps in the form
        //     <LastModified>2013-09-17T18:07:53.000Z</LastModified>
        // S3 GetObject returns timestamps in the form
        //            Last-Modified: Sun, 1 Jan 2006 12:00:00 GMT
```